tfhe/src/shortint/server_key/mod.rs

```diff
@@ -1463,15 +1453,42 @@ pub(crate) fn apply_programmable_bootstrap<InputCont, OutputCont>(
     bootstrapping_key: &ShortintBootstrappingKey,
     in_buffer: &LweCiphertext<InputCont>,
     out_buffer: &mut LweCiphertext<OutputCont>,
-    acc: &LookupTableOwned,
+    acc: GlweCiphertext<Vec<u64>>,
     buffers: &mut ComputationBuffers,
 ) where
     InputCont: Container<Element = u64>,
     OutputCont: ContainerMut<Element = u64>,
 {
-    let mut glwe_out = acc.acc.clone();
+    let mut glwe_out: GlweCiphertext<_> = acc;
```

Comment:
why change this ? it forces to add acc.acc.clone everywhere this function is called ?

Reply:
To avoid cloning in `apply_programmable_bootstrap`

This change is isolated on commit "refactor(shortint): apply_programmable_bootstrap takes a GlweCt"
There are only 4 places where adding this `.acc.acc.clone()` is needed

Let me know if you think it's still better to remove this change, it shouldn't affect following commits a lot

Reply:
we have to clone anyways, so keep it in this function please

Reply:
No, in some cases we have a owned lut which we use once so we don't have to clone it

Reply:
hmm

Reply:
maybe have a bound Into<GlweCiphertextOwned<u64>> ? and implement that for the &LookupTableOwned ?

Reply:
It would add complexity for not much IMO

Reply:
then keep the old behavior for now

Reply:
Done, but `apply_programmable_bootstrap` still take `&GlweCiphertextOwned` instead of `&LookupTableOwned` as the degree is not used at this level

Reply:
kind of defeat the purpose of the revert :/

Reply:
the thing with the generic is not so bad IMO